src/main/java/ch/njol/skript/events/EvtPlayerInput.java

```diff
@@ -0,0 +1,124 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerInputEvent;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.InputKey;
+
+import java.util.Set;
+
+public class EvtPlayerInput extends SkriptEvent {
+
+	static {
+		if (Skript.classExists("org.bukkit.Input")) {
+			Skript.registerEvent("Player Input", EvtPlayerInput.class, PlayerInputEvent.class,
+					"[player] (toggle|toggling|1:press[ing]|2:releas(e|ing)) [of] (%-inputkeys%|[a|any] key)",
+					"([player] %-inputkeys%|[a|any] [player] key) (toggle|toggling|1:press[ing]|2:releas(e|ing))")
+				.description("Called when a player sends an updated input to the server.",
+					"Note: The input keys event value is the set of keys the player is currently pressing, not the keys that were pressed or released.")
+				.examples("on player press any key:",
+					"\tsend \"You are pressing: %event-inputkeys%\" to player")
+				.since("INSERT VERSION")
+				.requiredPlugins("Minecraft 1.21.3+");
+		}
+	}
+
+	private @Nullable Literal<InputKey> keysToCheck;
+	private InputType type;
+
+	@Override
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
+		//noinspection unchecked
+		keysToCheck = (Literal<InputKey>) args[0];
+		type = InputType.values()[parseResult.mark];
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		PlayerInputEvent inputEvent = (PlayerInputEvent) event;
+		Set<InputKey> previousKeys = InputKey.fromInput(inputEvent.getPlayer().getCurrentInput());
+		Set<InputKey> currentKeys = InputKey.fromInput(inputEvent.getInput());
+		Set<InputKey> keysToCheck = this.keysToCheck != null ? Set.of(this.keysToCheck.getAll(event)) : null;
```

Comment:
```suggestion
		Set<InputKey> keysToCheck = this.keysToCheck != null ? Set.of(this.keysToCheck.getAll()) : null;
```
For a literal, you shouldn't need to pass the event.